src/server/controllers/UserTransaction.controller.ts

```diff
@@ -0,0 +1,175 @@
+import prisma from '@/lib/database/client';
+import { CreateUserDTO } from '@/lib/models/dto/CreateUser.dto';
+import { GetManyUserDTO } from '@/lib/models/dto/GetManyUser.dto';
+import { HashDTO } from '@/lib/models/dto/Hash.dto';
+import { IdDTO } from '@/lib/models/dto/Id.dto';
+import { getManyUserTransactionDtoSchema } from '@/lib/validations/GetManyUserTransactionDto.schema';
+import { hashDtoSchema } from '@/lib/validations/HashDto.schema';
+import { idDtoSchema } from '@/lib/validations/IdDto.schema';
+import { resolveBulkArgs } from '@/server/utils/resolveBulkArgs';
+import { resolvePrismaPaginationArgs } from '@/server/utils/resolvePrismaPaginationArgs';
+import { Prisma, UserTransaction } from '@prisma/client';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import { HttpResponseCodesEnum } from '../enums';
+import { InternalServerErrorException } from '../exceptions/InternalServerError.exception';
+import { NotFoundException } from '../exceptions/NotFound.exception';
+import { InfinitePaginationType } from '../types';
+import { resolveInfinitePagination } from '../utils/resolveInfinitePagination';
+import { resolveInfinitePaginationResponse } from '../utils/resolveInfinitePaginationResponse';
+import { validateSchema } from '../utils/validateSchema';
+import { CreateUserTransactionDTO } from '@/lib/models/dto/CreateUserTransaction.dto';
+import { createUserTransactionDtoSchema } from '@/lib/validations/CreateUserTransactionDto.schema';
+import { GetManyUserTransactionDTO } from '@/lib/models/dto/GetManyUserTransaction.dto';
+
+class UserTransactionController {
+  async create(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const body: CreateUserTransactionDTO = req.body;
+    validateSchema<CreateUserTransactionDTO>(createUserTransactionDtoSchema, body);
+
+    const response = await prisma.userTransaction.create({
+      data: {
+        contractHash: body.contractHash,
+        amount: body.amount,
+        userId: body.userId,
+      },
+    });
+
+    if (!response) {
+      throw new InternalServerErrorException('UserTransaction could not be created');
+    }
+
+    res.status(HttpResponseCodesEnum.CREATED).json(resolveInfinitePaginationResponse(response));
```

Comment:
Now we are only returning paginated response for getMany method, so you can return only response `.json(response)`